src/modules/party/party.service.ts

```diff
@@ -44,28 +49,135 @@ export class PartyService {
-          officerRiskDate: element.OfficerRiskDate ? element.OfficerRiskDate.slice(0, 10) : null,
-          countryCode: element.PrimaryAddress.Country.CountryCode,
-        })),
-      )
+      .then((acbsResponse) => acbsResponse.data)
       .catch((error) => {
         throw new GetPartiesBySearchTextException('Failed to get parties from ACBS.', error);
       });
 
     return response;
   }
 
+  async getPartiesBySearchText(token: string, searchText: string): Promise<GetPartiesBySearchTextResponse> {
+    const response = await this.getPartiesBySearchTextFromAcbs(token, searchText).then((partiesFromAcbs) =>
+      partiesFromAcbs.map((party) => ({
+        alternateIdentifier: party.PartyAlternateIdentifier,
+        industryClassification: party.IndustryClassification.IndustryClassificationCode,
+        name1: party.PartyName1,
+        name2: party.PartyName2,
+        name3: party.PartyName3,
+        smeType: party.MinorityClass.MinorityClassCode,
+        citizenshipClass: party.CitizenshipClass.CitizenshipClassCode,
+        officerRiskDate: party.OfficerRiskDate ? party.OfficerRiskDate.slice(0, 10) : null,
+        countryCode: party.PrimaryAddress.Country.CountryCode,
+      })),
+    );
+
+    return response;
+  }
+
+  async getPartyIdentifierBySearchText(token: string, searchText: string): Promise<CreatePartyResponse> {
+    const response = await this.getPartiesBySearchTextFromAcbs(token, searchText).then((partiesFromAcbs) => {
+      return partiesFromAcbs[0] ? (partiesFromAcbs[0].PartyIdentifier ? { partyIdentifier: partiesFromAcbs[0].PartyIdentifier } : {}) : undefined;
+    });
+
+    return response;
+  }
+
+  async createParty(token: string, party: CreatePartyRequestItem): Promise<{ partyIdentifier: string }> {
+    const acbsRequest = {
+      PartyAlternateIdentifier: party.alternateIdentifier,
+      IndustryClassification: {
+        IndustryClassificationCode: party.industryClassification,
+      },
+      PartyName1: party.name1,
+      PartyName2: party.name2 ?? '',
+      PartyName3: party.name3 ?? '',
+      PartyShortName: party.name1.substring(0, 15),
+      PartySortName: party.name1.substring(0, 20),
+      PartyType: {
+        PartyTypeCode: PROPERTIES.PARTY.DEFAULT.partyTypeCode,
+      },
+      PrimaryAddress: {
+        AddressIdentifier: PROPERTIES.PARTY.DEFAULT.address.addressIdentifier,
+        AddressName1: PROPERTIES.PARTY.DEFAULT.address.addressName1,
+        AddressType: {
+          AddressTypeCode: PROPERTIES.PARTY.DEFAULT.address.addressTypeCode,
+        },
+        Country: {
+          CountryCode: party.countryCode ?? PROPERTIES.PARTY.DEFAULT.address.countryCode,
+        },
+      },
+      DefaultGeneralLedgerUnit: {
+        GeneralLedgerUnitIdentifier: PROPERTIES.PARTY.DEFAULT.generalLedgerUnitIdentifier,
+      },
+      CitizenshipClass: {
+        CitizenshipClassCode: party.citizenshipClass,
+      },
+      OfficerRiskDate: `${party.officerRiskDate}T00:00:00Z`,
```

Comment:
Can we pick `T00:00:00Z` from constant since this has been used in other PRs as well. Thanks